File: C#.net/MTS.ICLViewer/frmClose.cs
//INSTANT C# NOTE: Formerly VB.NET project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;

namespace MTS.ICLViewer
{
    public partial class FrmClose
    {

        internal FrmClose()
        {
            InitializeComponent();
        }
        private void OK_Button_Click(object sender, System.EventArgs e)
        {
            if (cbClosePrompt.Checked)
            {
                Properties.Settings.Default.ClosePrompt = false;
                Properties.Settings.Default.Save();
            }
            this.DialogResult = System.Windows.Forms.DialogResult.Yes;
            this.Close();
        }

        private void Cancel_Button_Click(object sender, System.EventArgs e)
        {
            this.DialogResult = System.Windows.Forms.DialogResult.No;
            this.Close();
        }

    }

} //end of root namespace

File: VB.net/frmClose.vb
Imports System.Windows.Forms

Public Class frmClose

  Private Sub OK_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles OK_Button.Click
    If cbClosePrompt.Checked Then
      My.Settings.ClosePrompt = False
    End If
    Me.DialogResult = System.Windows.Forms.DialogResult.Yes
    Me.Close()
  End Sub

  Private Sub Cancel_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Cancel_Button.Click
    Me.DialogResult = System.Windows.Forms.DialogResult.No
    Me.Close()
  End Sub

End Class
